Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Generator.Equals.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            throw new Exception("Expected exactly one EquatableAttribute.");
+        }
+
+
+        var explicitMode = equatableAttributeData.GetNamedArgumentValue("Explicit") is true;
+        var ignoreInheritedMembers = equatableAttributeData.GetNamedArgumentValue("IgnoreInheritedMembers") is true;
+
+        if (_context.TargetSymbol is not ITypeSymbol symbol)
+        {
+            throw new Exception("Expected a type symbol.");
```

Comment:
Do not throw exceptions in your generator, this will just lead to bad user experience. The attribute is only valid on types; if it was misapplied, then just bail. The user will get a real compile error and there's nothing further for you to do.